### ISSUE TITLE
Add support for generating database indexes with MongoDB

### ISSUE THREAD

**Issue by graemerocher** [micronaut-projects/micronaut-data#1310]
### Feature description

See https://mongodb.github.io/mongo-java-driver/3.4/driver/tutorials/indexes/

**Comment by sowjanya-kotha** [comment]
Is this still an open issue?

**Comment by dstepanov** [comment]
Yes, we only support creating collections in `AbstractMongoCollectionsCreator`

**Comment by sowjanya-kotha** [comment]
Is there any documentation/guide that can help to create index along with ReactorCrudRepository
